src/main/java/HeadersRestTemplateService.java

```diff
@@ -0,0 +1,28 @@
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.stereotype.Service;
+
+@Service
+public class HeadersRestTemplateService {
+
+    @Value("${github.token}")
+    private String githubToken;
+
+    @Value("${openai.api.key}")
+    private String openaiApiKey;
+
+    public HttpHeaders createGitHubHeaders() {
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", "Bearer " + githubToken);
```

Comment:
Directly concatenating the token into the header could expose sensitive information if not handled properly. Ensure that the token is stored securely and not logged. (Severity: high)